Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -139,17 +163,81 @@ BTDiagnostics::InitializeFieldBufferData ( int i_buffer , int lev)
 {
     auto & warpx = WarpX::GetInstance();
     // 1. Lab-frame time for the i^th snapshot
+    m_t_lab[i_buffer] = i_buffer * m_dt_snapshots_lab;
```

Comment:
Just a thought: [`.at()`-access](https://en.cppreference.com/w/cpp/container/vector/at) is a bit "safer" for later code-restructuring as it throws an exception in performance uncritical code sections such as here, when accessed out of bounds.
```suggestion
    m_t_lab.at(i_buffer) = i_buffer * m_dt_snapshots_lab;
```